chain-indexer/src/infra/node/runtimes.rs

```diff
@@ -297,6 +311,36 @@ macro_rules! get_zswap_state_root {
 
 get_zswap_state_root!(runtime_0_13);
 
+macro_rules! get_transaction_cost {
+    ($module:ident) => {
+        paste::paste! {
+            async fn [<get_transaction_cost_ $module>](
+                online_client: &OnlineClient<SubstrateConfig>,
+                raw_transaction: &[u8],
+                block_hash: BlockHash,
+            ) -> Result<u128, SubxtNodeError> {
+                let get_transaction_cost = $module::apis()
+                    .midnight_runtime_api()
+                    .get_transaction_cost(raw_transaction.to_owned());
+
+                let (storage_cost, gas_cost) = online_client
+                    .runtime_api()
+                    .at(H256(block_hash.0))
+                    .call(get_transaction_cost)
+                    .await.map_err(Box::new)?
+                    .map_err(|error| SubxtNodeError::GetContractState(format!("Transaction cost error: {error:?}")))?;
```

Comment:
Move this "Transaction cost error" to the `#[error(...)]` and just pass the `error` here. Then clippy will tell you to replace the closure with `SubxtNodeError::GetContractState`.